src/libNetwork/P2PComm.cpp

```diff
@@ -654,6 +683,13 @@ void P2PComm::AcceptConnectionCallback([[gnu::unused]] evconnlistener* listener,
   bufferevent_enable(bev, EV_READ | EV_WRITE);
 }
 
+bool P2PComm::IsHostDownOrUnreachable() {
```

Comment:
Can reduce this to `return (errno == EHOSTUNREACH || errno == EHOSTDOWN);` and add `inline` to the function signature